tt_metal/common/bfloat8.hpp

```diff
@@ -248,12 +284,122 @@ inline std::vector<float> unpack_bfp8_tiles_into_float_vec(const std::vector<uin
 
                         uint32_t float_data_index;
                         if (row_major_output) {
-                            float_data_index = subtile_c + (32 * subtile_r) + (tile_index * num_float_in_tile);
+                            float_data_index = subtile_c + (tile_W * subtile_r) + (tile_index * num_float_in_tile);
                         } else {
                             float_data_index = fp32_element_index;
                             fp32_element_index += 8;
                         }
                         _mm256_storeu_ps(&float_vec[float_data_index],  _mm256_castsi256_ps(man));
+#else
+			std::vector<std::uint32_t> exp_vector(8, get_byte(exp_word, sub_word_index));
+                        std::vector<std::uint32_t> first(4, bfp8_tiles.at(16 + tile_and_data_index));
+                        std::vector<std::uint32_t> second(4, 16 + tile_and_data_index + 1);
+                        std::vector<std::uint32_t> combined(8, 0);
+
+                        for(std::size_t j = 0; j < 8; ++j) {
+			   if(shift[j] < 32) {
+   			      first[j] = first[j] & mask[j] >> shift[j];
+			   }
+			   else {
+   			      first[j] = 0;
+			   }
+			}
```

Comment:
It looks like this will perform out-of-bounds access on `first` (size == 4, loop iterates 8 times)

Reply:
tests/tt_metal/tt_metal/test_bfp8_conversion.cpp has a test case for the conversion code in bfloat8.hpp

Reply:
Nice catch, thanks for the code review. I'll make updates this evening.